### ISSUE TITLE
[Bug] Sometimes some package get error arch when change node version.

### ISSUE THREAD

**Issue by southorange0929** [volta-cli/volta#1481]
If change node version with `14.x` and `16.x` in apple m1. Some packages get wrong arch with `os.arch()` or `process.arch`.For example: `esbuild`

<img width="861" alt="图片" src="https://user-images.githubusercontent.com/32590310/229393333-d5b6f63e-7aca-4a3e-b99e-4cfd51d4daa5.png">

<img width="818" alt="图片" src="https://user-images.githubusercontent.com/32590310/229393357-4645b426-ce8b-4e33-a697-64af4d3badb2.png">

**Comment by fynnfluegge** [comment]
Hi @southorange0929,
I also faced this issue. The cause was that I ran `npm install` on node with x64 executable, hence packages for x64 were fetched, e.g. `esbuild`. I solved it by removing `esbuild` folder from `node_modules` and run `npm install` again. In my `node_modules` folder now `esbuild/darwin-arm64` was present and ready to use ✅

**Comment by southorange0929** [comment]
> Hi @southorange0929, I also faced this issue. The cause was that I ran `npm install` on node with x64 executable, hence packages for x64 were fetched, e.g. `esbuild`. I solved it by removing `esbuild` folder from `node_modules` and run `npm install` again. In my `node_modules` folder now `esbuild/darwin-arm64` was present and ready to use ✅

That's right.But if using `pnpm`,then removed `node_modules` and run `pnpm i`, get same error.

Maybe it's pnpm's bug?

**Comment by fynnfluegge** [comment]
Possible, pnpm has cold and hot cache and might load the deleted one again. You tried with npm?

**Comment by southorange0929** [comment]
I' try it later. Thanks :)
